test/models/requests/CreateUpdateExpensesRequestSpec.scala

```diff
@@ -0,0 +1,235 @@
+    otherAndCapitalAllowances = Some(210),
+    vehicleExpenses = Some(250),
+    mileageAllowanceRelief = Some(300)
+  )
+
+  val defaultModel: CreateUpdateExpensesRequest = CreateUpdateExpensesRequest(Some(true),
+    defaultExpenses)
+
+  val defaultPriorCustomerEmploymentExpenses: EmploymentExpenses = employmentExpenses.copy(
+    expenses = Some(defaultExpenses.copy(
+      businessTravelCosts = Some(15),
+      jobExpenses = Some(10),
+      flatRateJobExpenses = Some(5),
+      professionalSubscriptions = Some(14),
+      hotelAndMealExpenses = Some(12),
+      otherAndCapitalAllowances = Some(21),
+      vehicleExpenses = Some(25),
+      mileageAllowanceRelief = Some(30)
+    )))
+
+  "creates a create audit event" when {
+    "all expenses are provided" in {
+      val actualResult = defaultModel.toCreateAuditModel(taxYear)
+      val expectedJson = Json.parse(
+        s"""{
+           |	"taxYear": $taxYear,
+           |	"userType": "${user.affinityGroup.toLowerCase}",
+           |	"nino": "${user.nino}",
+           |	"mtditid": "${user.mtditid}",
+           |	"employmentExpensesData": {
+           |		"jobExpenses": 100,
+           |		"flatRateJobExpenses": 50,
+           |		"professionalSubscriptions": 140,
+           |		"otherAndCapitalAllowances": 210
+           |	}
+           |}""".stripMargin)
+      Json.toJson(actualResult) shouldBe expectedJson
+    }
+    "some expenses are provided  " in {
+      val model = defaultModel.copy(
+        expenses = defaultExpenses.copy(
+          professionalSubscriptions = None,
+          otherAndCapitalAllowances = None
+        )
+      )
+
+      val actualResult = model.toCreateAuditModel(taxYear)
+      val expectedJson = Json.parse(
+        s"""{
+           |	"taxYear": $taxYear,
+           |	"userType": "${user.affinityGroup.toLowerCase}",
+           |	"nino": "${user.nino}",
+           |	"mtditid": "${user.mtditid}",
+           |	"employmentExpensesData": {
+           |		"jobExpenses": 100,
+           |		"flatRateJobExpenses": 50
+           |	}
+           |}""".stripMargin)
+      Json.toJson(actualResult) shouldBe expectedJson
+    }
+    "no expenses are provided " in {
+      val model = defaultModel.copy(
+        expenses = defaultExpenses.copy(
+          jobExpenses = None,
+          flatRateJobExpenses = None,
+          professionalSubscriptions = None,
+          otherAndCapitalAllowances = None
+        )
+      )
+
+      val actualResult = model.toCreateAuditModel(taxYear)
+      val expectedJson = Json.parse(
+        s"""{
+           |	"taxYear": $taxYear,
+           |	"userType": "${user.affinityGroup.toLowerCase}",
+           |	"nino": "${user.nino}",
+           |	"mtditid": "${user.mtditid}",
+           |	"employmentExpensesData": {}
+           |}""".stripMargin)
+      Json.toJson(actualResult) shouldBe expectedJson
+    }
+  }
+  "creates an amend audit event" when {
+    "all prior expenses are provided" in {
```

Comment:
This is test could be named more accurately as "all prior and current expenses are provided"